tcms/bugs/tests/test_api.py

```diff
@@ -93,3 +93,27 @@ def verify_api_with_permission(self):
     def verify_api_without_permission(self):
         with self.assertRaisesRegex(ProtocolError, "403 Forbidden"):
             self.rpc_client.Bug.remove({"pk__in": [self.bug.pk, self.another_bug.pk]})
+
+
+class TestFilter(APITestCase):
+    """Test Bug.filter"""
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.bug = BugFactory(status=False)
+        self.another_bug = BugFactory(status=True)
+        self.yet_another_bug = BugFactory(status=True)
+
+    def test_filter(self):
+        result = self.rpc_client.Bug.filter({"status": True})
+        self.assertIsInstance(result, list)
+        self.assertEqual(len(result), 2)
+        # sort by pk of contained dicts
+        result.sort(key=lambda d: d["pk"])
+        self.assertEqual(result[0]["pk"], self.another_bug.pk)
+        self.assertEqual(result[1]["pk"], self.yet_another_bug.pk)
```

Comment:
The order is not guaranteed b/c the underlying query in .filter() doesn't sort the results. Use assertIn/assertNotIn to check the results or iterate over the resulting list to check things.